src/main/java/software/amazon/awssdk/s3accessgrants/plugin/internal/S3AccessGrantsUtils.java

```diff
@@ -47,6 +47,8 @@ public class S3AccessGrantsUtils {
 
     public static final Boolean DEFAULT_CROSS_REGION_ACCESS_SETTING = false;
 
+    public static String USER_AGENT = "aws-s3-accessgrants-java-sdk-v2-plugin";
```

Comment:
Is this useragent added to the call to the APIs that the plugin calls (like GetDataAccess /HeadObject etc) or to the data plane API that the customer is calling or both?

Reply:
"aws-s3-accessgrants-java-sdk-v2-plugin" is the actual artifact id of the plugin. We can add the version number here, but it will add an extra dependency if we want to do this dynamically. Doing this manually will need us to update this value every time we have a new revision. So unless its really necessary, I would skip that.

Reply:
> Is this useragent added to the call to the APIs that the plugin calls (like GetDataAccess /HeadObject etc) or to the data plane API that the customer is calling or both?

only to plugin calls like GetDataAccess, GetAccessGrantsInstanceForPrefix, HeadObject)